iree/compiler/Dialect/Flow/IR/FlowOps.cpp

```diff
@@ -491,6 +491,14 @@ void printDispatchRegionOp(OpAsmPrinter &p, DispatchRegionOp op) {
 // flow.executable
 //===----------------------------------------------------------------------===//
 
+DispatchEntryOp ExecutableOp::getDispatchEntryOp() {
```

Comment:
this isn't true, so it'd be nice not to add code that makes it impossible to do (this is part of hackability: someone should be able to put whatever they want in an executable and have it flow through the pipeline)

Reply:
We have this common structure today, without arbitrary IR from someone hacking at the pipeline:

```mlir
flow.executable ...
  flow.dispatch.entry
  module {
    func ... { }
  }
```

Deduping happens mainly at the `funcOp` level (since the entry point and function symbol names vary), with rewrites to symbol references down in `flow.dispatch`. If a pair of duplicate executables has multiple entry points and functions, that would need:
* the comparison between ops to handle multiple `funcOps`
* a loop over entry points when updating references

Those both seem reasonable, with an asterisk on ordering and comparison across executables (`foo` in `executable0` may match `bar` in `executable1`).